runtime/manifest-parser.peg

```diff
@@ -38,36 +41,186 @@ ManifestItem
   / Schema
 
 Import
-  = "import" whiteSpace path:id eolWhiteSpace
+  = 'import' whiteSpace path:id eolWhiteSpace
   {
     return {
-      kind: "import",
+      kind: 'import',
       path,
     }
   }
 
 Particle
-  = "particle" whiteSpace name:upperIdent implFile:(whiteSpace "in" whiteSpace id)? eolWhiteSpace body:IndentedText?
+  = 'particle' whiteSpace name:upperIdent implFile:(whiteSpace 'in' whiteSpace id)? eolWhiteSpace items:(Indent (SameIndent ParticleItem)*)? eolWhiteSpace?
   {
+    let args = null;
+    let affordance = [];
+    let slots = [];
+    let description = null;
+    items = items ? extractIndented(items) : [];
+    items.forEach(item => {
+      if (item.name == name) {
+        args = item.args;
+      } else if (item.kind == 'particle-slot') {
+        slots.push(item);
+      } else if (item.kind == 'description') {
+        description = {};
+        item.description.forEach(d => { description[d.name] = d.pattern; });
+      } else {
+        affordance.push(item.affordance)
+      }
+    });
+
     return {
-      kind: "particle",
+      kind: 'particle',
       name,
       implFile: optional(implFile, implFile => implFile[3], null),
-      body,
+      args,
+      affordance,
+      slots,
+      description
+    };
+  }
+
+ParticleItem
+  = ParticleInterface
+  / Affordance
+  / ParticleSlot
+  / ParticleDescription
+
+ParticleInterface
+  = name:upperIdent '(' args:ParticleArgumentList ')' eolWhiteSpace
```

Comment:
TODO: make the argument list optional

(to fix that bug that requires at least one particle argument)